include/seastar/core/sstring.hh

```diff
@@ -777,3 +778,23 @@ std::ostream& operator<<(std::ostream& os, const std::unordered_map<Key, T, Hash
     return os;
 }
 }
+
+namespace boost {
+    template<typename Target, typename Source>
+    inline Target lexical_cast(const Source& s);
+
+    template<>
+    inline seastar::sstring lexical_cast(const std::string& s) {
```

Comment:
How does this patch fix the problem with spaces in the string?

Reply:
Previously `boost::lexical_cast<seastar::sstring>` would crash with string with spaces (because of `sstring` it would fallback to generic implementation, which converts using streams, which read up to a white character).

This patch overrides the invalid Boost implementation (no generic stream implementation, just constructing `sstring` directly).

It fixes https://github.com/scylladb/scylla/issues/2905, because Boost command-line argument parsing implementation calls `lexical_cast<sstring>` internally - now it is replaced with correct implementation via full template specialization in functions.

Reply:
@tgrabiec ping.  Looks like we dropped the ball on this.

Reply:
There's a pending request: https://github.com/scylladb/seastar/pull/788/commits/85989cfb10ba26f5f28408a3c525b2b47e5e0f3f#r902560044